sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala

```diff
@@ -606,6 +606,36 @@ class StringFunctionsSuite extends QueryTest with SharedSparkSession {
       Seq(Row(Map("a" -> "1", "b" -> "2", "c" -> "3")))
     )
 
+    checkAnswer(
```

Comment:
Not really a test of my change, but it was a missing test (the case where a literal pair delimiter is given but no key/value delimiter is given).